### PR TITLE
feat: add typedoc

### LINKED ISSUE #79: Add Typedoc
## Description

To leverage the documentation in the code, add a documentation library, [typedoc](https://typedoc.org/).

Regarding issue: 
https://github.com/Tascurator/tascurator-frontend/issues/64

## Criteria

- [x] Install typedoc
- [x] update a scripts
- [x] Add a document deploy Actions

### PR DESCRIPTION
**Pull request by hitohata**
## Overview

Added a typedoc to share the document.

## Changes

- Added a typedoc
- Added a new script to run the typedoc, `npm run typedoc`
- Add GitHub Actions

## Review points

- I disabled the linter in this line. It has a type error. Please tell me how to fix it, or fix it after merging this PR. https://github.com/Tascurator/tascurator-frontend/blob/docs/79-add-typedoc/src/components/ui/datePicker.tsx#L41

## Screenshots or videos

image

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/datePicker.tsx [hunk 1]
```diff
@@ -38,7 +38,10 @@ export function DatePicker() {
         <Calendar
           mode="single"
           selected={date}
+          /* eslint-disable */
+          // @ts-ignore
           onSelect={setDate}
+          /* eslint-enable */
```
Comment (akiomatic) [inline src/components/ui/datePicker.tsx]: To fix the ESLint error, could you please make the setDate accept undefined by changing [this line](https://github.com/Tascurator/tascurator-frontend/blob/8735631bd9fb182c8330ce7365acc299c6bda188/src/components/ui/datePicker.tsx#L17)?

From
```ts
const [date, setDate] = useState<Date>(() => startOfDay(new Date()));
```
To
```ts
const [date, setDate] = useState<Date | undefined>(() => startOfDay(new Date()));
```

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you for adding TypeDoc, which I had never heard of.
I've left one comment to fix the ESLint error, so could you take a look at it?

**Comment by akiomatic** [comment]
@hitohata Also, I'd like you to change the PR title by making it similar to the commit messages. Thanks in advance.

```
PR Title Rules

Example:
- feat: add user authentication system
- fix: resolve navigation bar overlap issue
- refactor: improve error handling in payment gateway
- test(API): increase coverage for user authentication tests

```

**Comment by hitohata** [comment]
@akiomatic 
Thank you for your advice.
I have fixed it. Would you mind if I ask you to review this PR again, sir?

**Review by akiomatic — APPROVED** [review]
Thank you so much. LGTM. Don't call me sir...

**Pull request merged**